app/controllers/comments_controller.rb

```diff
@@ -1,20 +1,33 @@
 class CommentsController < ApplicationController
   before_action :authenticate_user!
```

Comment:
The before_actions are executed top-to-bottom and the first one to either:

   - redirect
   - render something

Causes the chain to stop.

Presumably your 404 test you didn't `sign_in` as a user, so the result was
a redirect to the login page, when you were hoping it to be a 404.  In your
test, if you sign in as a user, the then you could put it either place.

On Sat, Jan 2, 2016 at 3:47 PM, ceceliahavens <notifications@github.com>
wrote:

> In app/controllers/comments_controller.rb
> <https://github.com/FirehoseCommunity/DEFCON/pull/45#discussion_r48687348>
> :
>
> > @@ -1,20 +1,33 @@
> >  class CommentsController < ApplicationController
> >    before_action :authenticate_user!
>
> Thanks Ken! If I put it under the authenticate_user! it still doesn't
> work, but if I put it above it, it solves the problem. (I think I
> understand why that is, but maybe a little clarification from you wouldn't
> hurt?)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FirehoseCommunity/DEFCON/pull/45/files#r48687348>.
>



-- 
*Ken Mazaika*
Sprayer of the Firehose | co-founder
theFirehoseProject <http://www.thefirehoseproject.com/>

Need coding inspiration?  Find it here
<https://www.facebook.com/theFirehoseProject>!